### ISSUE TITLE
[9.3] Breaking change in recipe transfer handler slots

### ISSUE THREAD

**Issue by uecasm** [mezz/JustEnoughItems#2723]
Given a recipe transfer handler that is working in 9.2 and earlier, after upgrading to 9.3 it breaks by skipping over empty slots during enumeration (resulting in not leaving those slots empty and instead putting the next item there instead).

Testing in 9.3.3.95 with a vanilla quartz stair and clicking the + button, results in this:
![image](https://user-images.githubusercontent.com/539951/154230549-37f3e56c-3dc2-4703-9bfc-7558db328acb.png)
![image](https://user-images.githubusercontent.com/539951/154230676-9cda953e-c43e-434f-9217-20062c1fae08.png)

(It works ok in the vanilla crafting table; this is a backwards compatibility issue with a custom GUI.)

The transfer handler in question can be found here: https://github.com/ldtteam/minecolonies/blob/a084dc4c1c8ef4dd45018dccad44eda0714327cd/src/main/java/com/minecolonies/coremod/compatibility/jei/transfer/PrivateCraftingTeachingTransferHandler.java#L88
(I'm the author for the other mod.)

Note that it's assuming to see empty INPUT slots in the appropriate positions (and not just a sparse map -- but the above isn't even a sparse map since it has the "wrong" slot ids, if that's what the keys were intended to be).

**Comment by mezz** [comment]
Thanks for the report!
Backwards compatibility should be working, and I'd also expect this map to not be sparse.
This is a complex case, I will check out your code in a debugger and see what I can find out.

**Comment by mezz** [comment]
This should be fixed in JEI 9.4.1.97

**Comment by uecasm** [comment]
~~It does not appear to be fixed in that version.~~  Oh wait, I misread, let me try again.

Looks like it's working in 9.4.1.97.
